### PR TITLE
Applications packaged as not capable of loading modules contained within the archive

### PR DESCRIPTION
**Pull request by basz**
The ModuleAutoloader fails to load modules when the modules are part of the phar that contains the application.

A realpath check on phar://path/to/phar.phar/modules/Application/Module.php will return false. There no such thing as a realpath on the real filesystem for these kind of files.

This fix uses the Phar::current(true) method to find out the correct basepath of the application. Then regular loadModuleFromDir is after minor adjustment capable to load the module.php 'fromDir'. 

Note existing tests still pass, but I did not write a new one for this addition. I wouldn't know how to do that for such a case.

original #7148

### CODE REVIEW

**Comment by Ocramius** [comment]
Requiring a test case...

File: library/Zend/Loader/ModuleAutoloader.php [hunk 1]
```diff
@@ -232,12 +242,19 @@ public function autoload($class)
      */
     protected function loadModuleFromDir($dirPath, $class)
     {
-        $file = new SplFileInfo($dirPath . '/Module.php');
-        if ($file->isReadable() && $file->isFile()) {
+        $modulePath = $dirPath . '/Module.php';
+        if (substr($modulePath, 0, 7) == 'phar://') {
```
Comment (keradus) [inline library/Zend/Loader/ModuleAutoloader.php]: please use strict comparision

**Comment by basz** [comment]
@Ocramius : haz test

**Comment by Ocramius** [comment]
@basz thanks, looks much better!

**Comment by basz** [comment]
@Ocramius out of curiosity, if this lands, will it land in master when reviews are done or are you waiting with merging moments before 2.3.5 is released?

**Comment by Ocramius** [comment]
@basz I'm waiting for a legendary magical beast called "free time" ;-)

I merge when I have a bit of dedicated time to do so. Yes, merge will happen before 2.3.5/2.4

**Comment by basz** [comment]
fat

hook your magic method __sleep to __clone and time is multiplied.

thanks

**Comment by Ocramius** [comment]
@basz merged, thanks!

`master`: zendframework/zf2@d035b6ad93721497ae9314c168c9910c113a72a2
`develop`: zendframework/zf2@5aed8a9d77a8901d3abe9ee5f229dc1c26e817e7

**Pull request closed without merging**